workspaces/adventure-pack/goodies/typescript/Iterator.from/index.ts

```diff
@@ -0,0 +1,43 @@
+import "../Iterator.prototype.toIterable";
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  class Iterator<T> {
+    static from<TElem>(
+      object:
+        | Iterator<TElem>
+        | Iterable<TElem>
+        | { next(): IteratorResult<TElem> },
+    ): IterableIterator<TElem>;
+  }
```

Comment:
I think the snapshots can be unblocked after #419 and the following change:

```suggestion
  interface IteratorConstructor {
    from<TElem>(
      object:
        | Iterator<T>
        | Iterable<T>
        | { next(): IteratorResult<T> },
    ): IterableIterator<T>;
    
    const Iterator: IteratorConstructor;
  }
```

Reply:
thank you!